multinet/downloaders/d3_json.py

```diff
@@ -0,0 +1,77 @@
+"""Multinet downloader for nested JSON files."""
+import re
+
+from flasgger import swag_from
+
+from .. import db, util
+from ..errors import GraphNotFound
+
+from flask import Blueprint, make_response
+
+# Import types
+from typing import Any
+
+bp = Blueprint("download_d3_json", __name__)
+bp.before_request(util.require_db)
+
+
+@bp.route("/<workspace>/<graph>", methods=["GET"])
+@swag_from("swagger/d3_json.yaml")
+def download(workspace: str, graph: str) -> Any:
+    """Return a graph as a d3 json-encoded graph.
+
+    `workspace` - the target workspace
+    `graph` - the target graph
+    """
+
+    nodes = []
+    links = []
+
+    space = db.get_workspace_db(workspace)
+    if not space.has_graph(graph):
+        raise GraphNotFound(workspace, graph)
+
+    loaded_graph = space.graph(graph)
+
+    node_tables = loaded_graph.vertex_collections()
+    for node_table in node_tables:
+        table_nodes = list(loaded_graph.vertex_collection(node_table).all())
```

Comment:
Coercing the full collection to a list up front seems like a bad idea. Some graphs may have a *lot* of nodes.